tests/languages/android/ServiceTest.kt

```diff
@@ -21,28 +30,46 @@ import java.nio.file.Paths
 @Config(manifest=Config.NONE)
 @RunWith(AndroidJUnit4::class)
 class ServiceTest {
-    val filename: String = "result.txt"
+
+    private val filename: String = "result.txt"
 
     @Before
-    fun start() {
+    @ExperimentalCoroutinesApi
+    fun setUp() {
+        Dispatchers.setMain(Dispatchers.Unconfined)
         Files.deleteIfExists(Paths.get(filename))
         writeToFile("Test Started")
     }
 
+    @After
+    @ExperimentalCoroutinesApi
+    fun tearDown() {
+        Dispatchers.resetMain()
+    }
+
     @Test
     @Throws(IOException::class)
     fun test() {
-
         val client = Client(ApplicationProvider.getApplicationContext())
+            .setEndpointRealtime("wss://realtime.appwrite.org/v1")
+            .setProject("console")
+            .addHeader("Origin", "http://localhost")
+            .setSelfSigned(true)
         val foo = Foo(client)
         val bar = Bar(client)
         val general = General(client)
-        client.addHeader("Origin", "http://localhost")
-        client.setSelfSigned(true)
+        val realtime = Realtime(client)
+        var realtimeResponse = "Realtime failed!"
+
+        realtime.subscribe("tests") {
+            realtimeResponse = it
+                .toJson()
+                .fromJson(Map::class.java)["response"]!! as String
```

Comment:
Is it possible to skip the `toJson` and `fromJson` here and make it part of the internal callback?